cmake_uninstall.cmake.in

```diff
@@ -18,16 +18,25 @@ IF(NOT EXISTS "@CMAKE_CURRENT_BINARY_DIR@/install_manifest.txt")
   RETURN()
 ENDIF(NOT EXISTS "@CMAKE_CURRENT_BINARY_DIR@/install_manifest.txt")
 
+IF(NOT DEFINED PACKAGE_CREATES_DOT_CATKIN OR NOT "${PACKAGE_PREVIOUS_INSTALL_PREFIX}" EQUAL "${CMAKE_INSTALL_PREFIX}")
+  SET(PACKAGE_PREVIOUS_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}" CACHE INTERNAL "Cache install prefix given to the package")
+  IF(EXISTS "${CMAKE_INSTALL_PREFIX}/.catkin")
+    SET(PACKAGE_CREATES_DOT_CATKIN FALSE CACHE INTERNAL "")
+  ELSE()
+    SET(PACKAGE_CREATES_DOT_CATKIN TRUE CACHE INTERNAL "")
+  ENDIF()
+ENDIF()
+
 MESSAGE(STATUS "catkin path: @CMAKE_INSTALL_PREFIX@/.catkin")
-IF(EXISTS "@CMAKE_INSTALL_PREFIX@/.catkin")
+IF(EXISTS "@CMAKE_INSTALL_PREFIX@/.catkin" AND NOT PACKAGE_CREATES_DOT_CATKIN)
```

Comment:
Actually we want the opposite condition here
```suggestion
IF(EXISTS "@CMAKE_INSTALL_PREFIX@/.catkin" AND PACKAGE_CREATES_DOT_CATKIN)
```

Reply:
If I implement the suggested change, then cmake uninstall the .catkin file.
I could change the name of the variable if it confuses.

Reply:
That seems strange, in the scenario you described you should have `PACKAGE_CREATES_DOT_CATKIN` set to `FALSE` and in that case we should not delete the file so, unless I misunderstood something about the scenario, the suggested change should be correct.

Note that because we are interacting with the cache it might be easy to get mixed up while testing this on incremental runs of the changes.

Reply:
@gergondet -- please let me know what are your thoughts! Looking forward to fixing this across different projects